core/src/main/java/tachyon/worker/hierarchy/StorageDir.java

```diff
@@ -105,91 +115,155 @@
    * @param blockId Id of the block
    */
   public void accessBlock(long blockId) {
-    mLastBlockAccessTimeMs.put(blockId, System.currentTimeMillis());
+    if (containsBlock(blockId)) {
+      mLastBlockAccessTimeMs.put(blockId, System.currentTimeMillis());
+    }
   }
 
   /**
-   * Add information of a block in current StorageDir
+   * Adds a block into the dir.
+   * 
+   * @param blockId the Id of the block
+   * @param sizeBytes the size of the block in bytes
+   * @param report need to be reported during heartbeat with master
+   */
+  private void addBlock(long blockId, long sizeBytes, boolean report) {
+    addBlock(blockId, sizeBytes, System.currentTimeMillis(), report);
+  }
+
+  /**
+   * Adds a block into the dir.
    * 
    * @param blockId Id of the block
    * @param sizeBytes size of the block in bytes
+   * @param accessTimeMs access time of the block in millisecond.
+   * @param report whether need to be reported During heart beat with master
    */
-  private void addBlockId(long blockId, long sizeBytes) {
-    accessBlock(blockId);
+  private void addBlock(long blockId, long sizeBytes, long accessTimeMs, boolean report) {
```

Comment:
I think it is not very accurate to name it addBlock, since it only add information of the block into StorageDir,  and cacheBlock actually adds the block.  the corresponding methods for removing blocks are deleteBlock and deleteBlockId.